test/documents.spec.js

```diff
@@ -0,0 +1,140 @@
+      peerId1 = await getIpfsPeerId(ipfs1)
+      peerId2 = await getIpfsPeerId(ipfs2)
+
+      keystore = new Keystore('./keys_1')
+      await keystore.open()
+      for (const [key, value] of Object.entries(identityKeys)) {
+        await keystore.addKey(key, value)
+      }
+
+      signingKeystore = new Keystore('./keys_2')
+      await signingKeystore.open()
+      for (const [key, value] of Object.entries(signingKeys)) {
+        await signingKeystore.addKey(key, value)
+      }
+
+      // Create an identity for each peers
+      testIdentity1 = await createIdentity({ id: 'userA', keystore, signingKeystore })
+      testIdentity2 = await createIdentity({ id: 'userB', keystore, signingKeystore })
+      
+      const accessController = {
+        canAppend: (entry) => entry.identity.id === testIdentity1.id
+      }
+    })
+    
+    beforeEach(async () => {
+      db1 = await Documents({ OpLog: Log, Database, ipfs: ipfs1, identity: testIdentity1, databaseId, accessController })
+    })
+
+    afterEach(async () => {
+      if (db1) {
+        await db1.drop()
+        await db1.close()
+      }
+      if (db2) {
+        await db2.drop()
+        await db2.close()
+      }
+    })
+
+    after(async () => {
+      if (ipfsd1) {
+        await stopIpfs(ipfsd1)
+      }
+      if (ipfsd2) {
+        await stopIpfs(ipfsd2)
+      }
+      if (keystore) {
+        await keystore.close()
+      }
+      if (signingKeystore) {
+        await signingKeystore.close()
+      }
+      if (testIdentity1) {
+        rmrf(testIdentity1.id)
+      }
+      if (testIdentity2) {
+        rmrf(testIdentity2.id)
+      }
+      rmrf('./orbitdb')
+      rmrf('./keys_1')
+      rmrf('./keys_2')
+    })
+
+    describe('using database', () => {
+      it('gets a document', async () => {
+         const key = 'hello world 1'
+
+         await db1.put({ _id: key, doc: 'writing 1 to db1' })
+         
+         const doc = await db1.get(key)
+         strictEqual(doc._id, key)
+      })
+      
+      it('deletes a document', async () => {
+         const key = 'hello world 1'
+         
+         await db1.put({ _id: key, doc: 'writing 1 to db1' })
+         await db1.del(key)
+         
+         const doc = await db1.get(key)
+         strictEqual(doc, null)
+      })
+
+      it('queries a document', async () => {
+          const expected = { _id: 'hello world 1', doc: 'writing new 1 to db1', views: 10 }
+          
+          const doc3 = { _id: 'hello world 3', doc: 'writing 3 to db1', views: 12 }
+          
+          await db1.put({ _id: 'hello world 1', doc: 'writing 1 to db1', views: 10 })
+          await db1.put({ _id: 'hello world 2', doc: 'writing 2 to db1', views: 5 })
+          await db1.put({ _id: 'hello world 3', doc: 'writing 3 to db1', views: 12 })
+          await db1.del('hello world 3')          
+          await db1.put(expected)
+          
+          deepStrictEqual(await db1.query((e) => e.views > 5), [expected])
```

Comment:
Small thing, but it'd be nice to put the `mapper` function here into it's own variable, eg. `filterFn` as per previous comment, so that it makes it a bit more clear what the user should pass into the `query` function.